libs/parallelism/algorithms/include/hpx/parallel/task_group.hpp

```diff
@@ -0,0 +1,110 @@
+//  Copyright (c) 2021 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file task_group.hpp
+
+#pragma once
+
+#include <hpx/config.hpp>
+
+#include <hpx/assert.hpp>
+#include <hpx/concepts/concepts.hpp>
+#include <hpx/datastructures/tuple.hpp>
+#include <hpx/errors/exception_list.hpp>
+#include <hpx/execution/traits/is_executor.hpp>
+#include <hpx/execution_base/execution.hpp>
+#include <hpx/executors/parallel_executor.hpp>
+#include <hpx/functional/invoke_fused.hpp>
+#include <hpx/synchronization/latch.hpp>
+
+#include <stdexcept>
+#include <type_traits>
+#include <utility>
+
+namespace hpx { namespace execution { namespace experimental {
+
+    ///////////////////////////////////////////////////////////////////////////
+    class task_group
+    {
+    public:
+        task_group()
+          : latch_(1)
+        {
+        }
+
+        ~task_group()
+        {
+            // wait() must have been called
+            HPX_ASSERT(latch_.is_ready());
+        }
+
+    public:
+        /// Spawns a task to compute f() and returns immediately.
+        // clang-format off
+        template <typename F, typename... Ts,
+            HPX_CONCEPT_REQUIRES_(
+                !hpx::traits::is_executor_any<std::decay_t<F>>::value
+            )>
+        // clang-format on
+        void run(F&& f, Ts&&... ts)
+        {
+            latch_.count_up(1);
+            hpx::parallel::execution::post(execution::parallel_executor{},
+                [this, f = std::forward<F>(f),
+                    t = hpx::forward_as_tuple(
+                        std::forward<Ts>(ts)...)]() mutable {
+                    try
+                    {
+                        hpx::util::invoke_fused(std::move(f), std::move(t));
+                    }
+                    catch (...)
+                    {
+                        errors_.add(std::current_exception());
```

Comment:
This needs to do something like https://github.com/STEllAR-GROUP/hpx/blob/87c0d769cd627a70c1e1d81a63b408b68a952c26/libs/parallelism/futures/include/hpx/futures/detail/future_data.hpp#L531-L548 (because of the lock inside `exception_list::add`) or ideally use https://github.com/STEllAR-GROUP/hpx/blob/93714c3e77bdcccc0097a096631c7280cfe6bd31/libs/core/execution_base/include/hpx/execution_base/detail/try_catch_exception_ptr.hpp from #5274.